src/ln/channelmonitor.rs

```diff
@@ -286,12 +286,27 @@ impl ChannelMonitor {
 				}
 			}
 		}
-		// TODO: Prune payment_preimages no longer needed by the revocation (just have to check
-		// that non-revoked remote commitment tx(n) do not need it, and our latest local commitment
-		// tx does not need it.
+
+		let mut pruned_payment_preimages = HashMap::with_capacity(self.payment_preimages.capacity());
+		for (k, v) in self.payment_preimages.drain() {
+			for (htlc, _s1, _s2) in &self.current_local_signed_commitment_tx.as_ref().expect("Channel need at least an initial commitment tx !").htlc_outputs {
+				if k == htlc.payment_hash {
+					pruned_payment_preimages.insert(k, v); 
+				}
+			}
+			for (_tx, htlcs) in &self.remote_claimable_outpoints {
+				for htlc in htlcs {
+					if k == htlc.payment_hash {
+						pruned_payment_preimages.insert(k, v);
+					}
+				}
+			}
+		}
+		self.payment_preimages = pruned_payment_preimages;
+
 		Ok(())
 	}
-
+	
```

Comment:
Note you have some trailing whitespace in a few places.

Reply:
Yes, configured my vim extension accordingly